### PR TITLE
remove the l10n  locale list match validate  

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 1]
```diff
@@ -31,11 +31,13 @@
 	
 	private static Logger LOGGER = LoggerFactory.getLogger(CollectSourceValidationInterceptor.class);
 	
-	public CollectSourceValidationInterceptor(List<String> sourceLocales, Map<String, List<String>> allowListMap) {
+	public CollectSourceValidationInterceptor(List<String> sourceLocales, Map<String, List<String>> allowListMap, String sourceLocalesStr) {
```
Comment (chaoq0318) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:34]: Is it possible to use sourceLocales directly without adding a new parameter: sourceLocalesStr?

**Review by linr211 — COMMENTED** [review]
Is there a bug description or feature requirement for this? I don't fully understand the purpose of this PR.

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 2]
```diff
@@ -31,11 +31,13 @@
...
 		this.sourceLocales = sourceLocales;
 		this.allowList = allowListMap;
+		this.sourceLocalesStr = sourceLocalesStr;
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:37]: Need to know the purposes of sourceLocales and sourceLocaleStr.

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 3]
```diff
@@ -181,8 +183,9 @@ private static void validateLocale(HttpServletRequest request, List<String> sour
 		}
 		if (!RegExpValidatorUtils.IsLetterAndNumberAndValidchar(locale)) {
 			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
-		}else if (!sourceLocales2.contains(locale)) {
-			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
+		}
+		if (!sourceLocales2.get(0).equals(ConstantsKeys.FALSE) && !sourceLocales2.contains(locale)) {
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:187]: It's strange to use argument of index 0 to judge.

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java (hunk 2 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:37]: sourceLocales use validate the collect resource  locale， sourceLocaleStr use response String when validate false

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java (hunk 1 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:34]: done

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java (hunk 3 above)
Comment (tigershi) [reply g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:186]: the list interface not support get first method

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 4]
```diff
@@ -181,11 +182,24 @@ private static void validateLocale(HttpServletRequest request, List<String> sour
...
+			throw new VIPAPIException(String.format(ValidationMsg.LOCALENAME_ONLY_SUPPORTED, getSourceSupportLocales(sourceLocales2)));
 		}
 	}
 
+	private static String getSourceSupportLocales(List<String> sourceLocaleList) {
+		if(sourceLocaleList == null || sourceLocaleList.size() == 0) {
+			return "";
+		}
+		StringBuilder sb = new StringBuilder();
```
Comment (chaoq0318) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:195]: Maybe we can use `StringUtils.join` directly, such as
`return StringUtils.join(sourceLocaleList, ConstantsChar.COMMA + ConstantsChar.SPACING);`

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 5]
```diff
@@ -181,11 +182,24 @@ private static void validateLocale(HttpServletRequest request, List<String> sour
 		}
 		if (!RegExpValidatorUtils.IsLetterAndNumberAndValidchar(locale)) {
 			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
-		}else if (!sourceLocales2.contains(locale)) {
-			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
+		}
+		if (!sourceLocales2.get(0).equals(ConstantsKeys.FALSE) && !sourceLocales2.contains(locale)) {
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:186]: If sourceLocales2 is empty, don't need to verify the locale. Is this more reasonable?

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 6]
```diff
@@ -181,11 +182,24 @@ private static void validateLocale(HttpServletRequest request, List<String> sour
...
+			throw new VIPAPIException(String.format(ValidationMsg.LOCALENAME_ONLY_SUPPORTED, getSourceSupportLocales(sourceLocales2)));
 		}
 	}
 
+	private static String getSourceSupportLocales(List<String> sourceLocaleList) {
```
Comment (Xiaochao8) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:191]: To improve performance, it's better to prepare this when application starts up.

**Review by Xiaochao8 — APPROVED** [review]

File: g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java [hunk 7]
```diff
@@ -19,6 +19,7 @@
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.vmware.vip.api.rest.APIParamName;
+import com.vmware.vip.common.constants.ConstantsChar;
```
Comment (chaoq0318) [inline g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:22]: Is this used?

Comment (tigershi) [reply g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java:22]: yes

**Review by chaoq0318 — APPROVED** [review]

**Pull request merged**